lib/metasploit_payloads/mettle.rb

```diff
@@ -176,8 +176,21 @@ def self.available_extensions(platform)
       end
 
       extensions = ::Dir.entries(dir_path)
+      extensions.select! { |extension| !extension.end_with?('.bin') }
+      # The stdapi extension is baked in
+      extensions << 'stdapi'
       # Only return extensions!
-      extensions - [ '.', '..', 'mettle', 'mettle.bin' ]
+      extensions -= [ '.', '..', 'mettle', 'mettle' ]
+      extensions
+    end
+
+    #
+    # List platforms on which the specified extension is available for loading
+    #
+    def self.available_platforms(extension)
+      ::Dir.entries(path).select do |platform|
```

Comment:
Um question but where is `path` coming in here? I don't see it defined anywhere.